src/imperator/tables/effects.rs

```diff
@@ -0,0 +1,281 @@
+use crate::effect::Effect;
+use crate::everything::Everything;
+use crate::item::Item;
+use crate::scopes::*;
+use crate::token::Token;
+use crate::imperator::effect_validation::{EvB, EvBv, EvV};
+
+use Effect::*;
+
+pub fn scope_effect(name: &Token, _data: &Everything) -> Option<(Scopes, Effect)> {
+    let lwname = name.as_str().to_lowercase();
+
+    for (from, s, effect) in SCOPE_EFFECT {
+        if lwname == *s {
+            return Some((Scopes::from_bits_truncate(*from), *effect));
```

Comment:
After replacing `u64` with `Scopes` below, this line will have just `*from` instead of `Scopes::from_bits_truncate(*from)`